python/renderer_status.py

```diff
@@ -0,0 +1,32 @@
+import requests
+from urllib.parse import urljoin
+
+def main(args):
+
+    # Tell the API the current render sttus
+    try:
+        # get passed-in arguments
+        api_url = args['CHOIRLESS_API_URL']
+        api_key = args['CHOIRLESS_API_KEY']
+        choir_id = args['choir_id']
+        song_id = args['song_id']
+        part_id = args['part_id']
```

Comment:
On the latter pipeline stages, we won't have a `part_id` so we need to make getting this optional and deal with it not being there later. Does the API work if it is not supplied?

Reply:
Ah this is a problem. The triumvirate of choirId/songId/partId are the key to tracking _this_ render job throughout its life in the pipeline. As such, the API requires all three.

Reply:
Obvs we can modify the API to key on choidId/songId, but you could get two overlapping renders posting conflicting statuses running at the same time. Maybe that's ok as it's a rare event?

Reply:
modified the API to take part_id optionally.

This Python doesn't insist on part_id.